core/src/main/java/io/aiven/klaw/service/OperationalRequestsService.java

```diff
@@ -0,0 +1,341 @@
+              .sorted(getPreferredOrdering(order))
+              .collect(Collectors.toList());
+    } catch (Exception e) {
+      log.error("No environments/clusters found.", e);
+      return new ArrayList<>();
+    }
+    return topicReqs;
+  }
+
+  private Comparator<OperationalRequest> getPreferredOrdering(Order order) {
+    return switch (order) {
+      case ASC_REQUESTED_TIME -> compareByTime();
+      case DESC_REQUESTED_TIME -> Collections.reverseOrder(compareByTime());
+    };
+  }
+
+  private static Comparator<OperationalRequest> compareByTime() {
+    return Comparator.comparing(OperationalRequest::getRequesttime);
+  }
+
+  public ApiResponse resetConsumerOffsets(String req_no) {
+    log.info("approveConsumerOffsetRequests {}", req_no);
+    final String userDetails = getUserName();
+    int tenantId = commonUtilsService.getTenantId(userDetails);
+    if (commonUtilsService.isNotAuthorizedUser(
+        getPrincipal(), PermissionType.APPROVE_SUBSCRIPTIONS)) {
+      return ApiResponse.NOT_AUTHORIZED;
+    }
+    ResetConsumerGroupOffsetsRequest resetConsumerGroupOffsetsRequest =
+        ResetConsumerGroupOffsetsRequest.builder().build();
+
+    HandleDbRequests dbHandle = manageDatabase.getHandleDbRequests();
+    OperationalRequest operationalRequest =
+        dbHandle.getOperationalRequest(Integer.parseInt(req_no), tenantId);
+    ApiResponse apiResponse;
+    try {
+      apiResponse =
+          clusterApiService.resetConsumerOffsets(
+              resetConsumerGroupOffsetsRequest, operationalRequest.getEnvironment(), tenantId);
+    } catch (KlawException e) {
+      return ApiResponse.notOk(ApiResultStatus.FAILURE.value);
+    }
+
+    if (apiResponse.isSuccess()) {
+      if (apiResponse.getData() instanceof Map) {
+        Map<OffsetsTiming, Map<String, Long>> offsetPositionsBeforeAndAfter =
+            (Map) apiResponse.getData();
+        String beforeReset =
+            "\n\nBefore Offset Reset"
+                + offsetPositionsBeforeAndAfter.get(OffsetsTiming.BEFORE_OFFSET_RESET);
+        String afterReset =
+            "\n\nAfter Offset Reset"
+                + offsetPositionsBeforeAndAfter.get(OffsetsTiming.AFTER_OFFSET_RESET);
+        String offsetResetDetails =
+            resetConsumerGroupOffsetsRequest.getConsumerGroup() + "\n" + beforeReset + afterReset;
+        mailService.sendMail(
+            operationalRequest.getTopicname(),
+            offsetResetDetails,
+            "",
+            operationalRequest.getRequestor(),
+            operationalRequest.getApprover(),
+            operationalRequest.getRequestingTeamId(),
+            dbHandle,
+            MailType.RESET_CONSUMER_OFFSET_APPROVED,
+            commonUtilsService.getLoginUrl());
+      }
+    }
+
+    return apiResponse.isSuccess()
+        ? ApiResponse.ok(ApiResultStatus.SUCCESS.value)
+        : ApiResponse.notOk(ApiResultStatus.FAILURE.value);
```

Comment:
Since there are already existing constants it could be shorter
```suggestion
    return apiResponse.isSuccess() ? ApiResponse.SUCCESS: ApiResponse.FAILURE;
```